src/lib/replication/service/raft_repl_service.cpp

```diff
@@ -156,17 +156,20 @@ void RaftReplService::start() {
     LOGINFO("Starting DataService");
     hs()->data_service().start();
 
-    // Step 6: Iterate all the repl dev and ask each one of the join the raft group.
-    for (auto it = m_rd_map.begin(); it != m_rd_map.end();) {
-        auto rdev = std::dynamic_pointer_cast< RaftReplDev >(it->second);
-        rdev->wait_for_logstore_ready();
-        if (!rdev->join_group()) {
-            HS_REL_ASSERT(false, "FAILED TO JOIN GROUP, PANIC HERE");
-            it = m_rd_map.erase(it);
-        } else {
-            ++it;
-        }
-    }
+    // Step 6: Iterate all the repl devs and ask each one of them to join the raft group concurrently.
```

Comment:
Looks fine to me to change to join raft group concurrently. Since this is some kind of behavior change, it probably worth to mention to the team and be aware of any testing outcomes.

Reply:
ack